### DISCUSSION TITLE
Reading an inactive field on sparse snode sometimes have non-zero value.

### DISCUSSION THREAD

**Discussion by identxxy** [taichi-dev/taichi#7684] [Q&A]
When I do for loop across a sparse field, which is like projection, I can get 
<img width="190" alt="image" src="https://user-images.githubusercontent.com/32609259/228494254-088ac3b4-1ee3-4a83-928f-680e4459f2ef.png">

However, when I manually query the field, which is like ray marching, I get
<img width="210" alt="image" src="https://user-images.githubusercontent.com/32609259/228494598-2d47007f-ced3-4206-9d02-c817294a687f.png">

This issue only occur on some sparse field, not all, but it is hard to find a certain simple case with this issue...
Any help or suggestion why this could happen?

Here is a clearer demo  of the issue. The `basis` is a field with the hierarchy `root => pointer => pointer => pointer=>dense=>field`.
There is no deactivation on the sparse field. The activation should also be in correct order.

The kernel to check activation
``` python
@ti.kernel
def check_activation(self, f: ti.template(), xi: ti.i32, yi:ti.i32, zi:ti.i32) -> bool:
    ancestor_idx = ti.rescale_index(f, f.parent(2), [xi, yi, zi])
    return ti.is_active(f.parent(2), ancestor_idx)
```

Printing in gui
``` python
                voxel = basis[self._debug_voxel[0], self._debug_voxel[1], self._debug_voxel[2]]
                gui.text(f"voxel : {voxel.sigma} {voxel.color}")
                activation = self.check_activation(basis,
                                                   self._debug_voxel[0],
                                                   self._debug_voxel[1],
                                                   self._debug_voxel[2],
                                                   )
                gui.text(f"activation : {activation}")
```

<img width="132" alt="image" src="https://user-images.githubusercontent.com/32609259/228492560-618eae08-e9db-4bf5-a0f0-b6330ad49e3a.png">

Here the inactive voxel also have non-zero values.

**Comment by identxxy** [comment]
The above screenshot is taichi 1.5.0 with `ti.cuda` backend.
`[Taichi] version 1.5.0, llvm 15.0.1, commit 7b885c28, win, python 3.9.13`

If I go back to 1.2.1,
 `[Taichi] version 1.2.1, llvm 10.0.0, commit 12ab828a, win, python 3.9.13`
it has the correct result.

<img width="338" alt="image" src="https://user-images.githubusercontent.com/32609259/228518181-29a2087f-04ce-4648-9d40-a48d3268a52b.png">

**Comment by ailzhang** [comment]
@identxxy is it possible to have a minimal repro for this issue so that I can run on my side? Thanks!
